### PR TITLE
Add github actions and test

### PR DESCRIPTION
**Pull request by nilsnolde**
- migrated image builds to github actions
- kept dockerhub as repo
- only tag pushes will actually push the image to dockerhub
- added a test that's triggered on tag pushes & pull requests

and sorry @jcoupey , those env vars were just half the story.. you'd also need to generate a token on dockerhub that you can use as the docker password env: https://github.com/docker/login-action#docker-hub. would've expected that your _actual_ password works, but it's safer this way anyways I guess.. 

can you let me know when you did that? once that's done I'll push a v1.11.0-rc.1 tag and see if it pushes to dockerhub alright and we can merge (apart from general review)..

the test already went through: https://github.com/VROOM-Project/vroom-docker/runs/3805859473?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/build_test_publish.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# only builds and tests on tag pushes and PRs to master
+# only pushes on tag pushes
+name: build, test, publish
+on:
+  push:
+    branches:
+      - 'master'
+      - 'add-gh-actions'
```
Comment (nilsnolde) [inline .github/workflows/build_test_publish.yml:8]: just for initial testing, next commit will remove it again, now that it's a PR

File: tests/test.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
+set -u
+
+CONF_DIR="${PWD}/conf"
+PORT=3005
+
+# save the query to eval later
+status_request="curl -X POST -H 'Content-Type: application/json' -d @tests/test_req.json http://localhost:${PORT}"
```
Comment (nilsnolde) [inline tests/test.sh:7]: just requests the healthcheck example, included in this repo

**Review by jcoupey — APPROVED** [review]
Looks good as far as my limited GitHub Actions experience can tell. I did replace the `DOCKER_PASSWORD` secret with a token generated from Docker Hub for user `DOCKER_USERNAME`.

**Comment by nilsnolde** [comment]
@jcoupey doesn't quite work yet for some reason.. don't think it's the github action, I guess there's still a problem with the secrets:
https://github.com/VROOM-Project/vroom-docker/runs/3812883690?check_suite_focus=true

ref docker/login: https://github.com/docker/login-action#docker-hub

**Comment by nilsnolde** [comment]
what's your username? you need to use the docker ID (for me nilsnolde), not the email (I think you log in on dockerhub.com with email, but you can't authenticate with it, dockerhub is a mess)

**Comment by nilsnolde** [comment]
you can try in your terminal if it works with the values: `docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"`

**Comment by jcoupey** [comment]
I was using docker id all right (not email). I just reset the `DOCKER_USERNAME` and `DOCKER_PASSWORD` values using id and token and it seems to work after re-running the job?

The fun (!?) thing is that using the same credentials in the above `docker login` command fails (permission denied).

**Comment by jcoupey** [comment]
And the RC tag is now available on Docker hub: https://hub.docker.com/layers/vroomvrp/vroom-docker/1.11.0-rc.1/images/sha256-f9fb835cc471676d39aaa0b89c16c8c2e2caa8c545e445ed4e05806e1c9e6d0c?context=explore

Nit: this tag breaks the convention from previous versions, it should be v1.11.0-rc.1

**Comment by jcoupey** [comment]
In related news: looks like GitHub only relies on diffs to decide if a PR is merged, regardless of whether the branch has actually been git-merged. A bit misleading at first!

**Comment by nilsnolde** [comment]
great!

> Nit: this tag breaks the convention from previous versions, it should be v1.11.0-rc.1

that's then a bug in https://github.com/docker/metadata-action#typesemver, where it says `tags: type=semver,pattern={{raw}}` would simply use the raw tag name without strictly converting to semantic versioning. I'll use `tags: type=semver,pattern=v{{raw}}` then. 

you wanna remove the docker image and I'll force push the tag again with the fix above so it's consistent?

**Comment by jcoupey** [comment]
> you wanna remove the docker image from dockerhub and I'll force push the tag again with the fix above so it's consistent?

Yes, I just deleted it.

**Comment by nilsnolde** [comment]
jeez.. https://github.com/VROOM-Project/vroom-docker/runs/3813717695?check_suite_focus=true

the metadata action still thinks the `v` should be dropped.. that's why I hate these blackbox plugin tools which are basically untestable.. I'll postpone this for now and get back to it on the weekend

**Pull request merged**